### ISSUE TITLE
Extend table API to support duplicating tables

### ISSUE THREAD

**Issue by kgodey** [centerofci/mathesar#453]
## Problem

The design spec for [Basic Table Operations](https://wiki.mathesar.org/design/specs/table-operations) calls for the ability to duplicate a table.

## Proposed solution

The Table creation endpoint should be extended to accept a reference to another table and whether all data should be copied. If this parameter is passed in, it should create a new table that's a copy of the table that has been passed in. Data should be copied conditionally according to the parameter that has been passed in.

## Additional context

- #185 
- Mathesar's [API Standards](https://wiki.mathesar.org/en/engineering/architecture/api-standards)

**Comment by bohemia420** [comment]
@kgodey there can be 3 possibilities IMO, 
- clone (schema, constraints only) clone (schema, constraints+data)
- (deep) copy

agree? or is it just clone and reference, to begin with?

**Comment by kgodey** [comment]
@bohemia420 What do you anticipate being copied in a "deep copy" that isn't covered by the first two options?

**Comment by bohemia420** [comment]
say a table A with shape -[100,4] i.e 100 rows 4 cols. creating a table B could be:
- table B shaped [0, 4] i.e just copied the schema.
- table B shaped [100, 4] i.e copied schema + data. Should there be filters/parameters, shape may be [x, 4] where x <= 100
these are the first two options.

the "deep copy" may actually be more of a "view", which albeit [100, 4] or as per a filter say [x, 4] where x <= 100, **but any changes to table B will also be reflected in table A**. IMO, I don't think much such cases. One real life has been an `alerts table` primarily comprising **ALARMS** and **STATUS OKs**. 
we had had a projection (`type=ALARM`) so we only look at ALARMs in an otherwise table of "STATUS OKs", and as we'd resolve the ALARMs (over a UI), both these tables A and B 'd get updated. 

TL;DR - honoring simplicity, we should be good with first two cases only.

**Comment by kgodey** [comment]
@bohemia420 We'd want users to make a view for cases where they want data automatically updated from other tables, see #442 for issues related to that. This issue already covers the first two options so we should be good.

**Comment by kgodey** [comment]
I'm going to close this, we can reopen or open a new issue once we have a more detailed design for it.
